src/components/IDO/Ido.tsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react'
+import { Grid, Typography, Box, Button } from '@material-ui/core'
+import { IdoStats } from './IdoStats'
+import ExchangeAmountInput from '@components/Inputs/ExchangeAmountInput/ExchangeAmountInput'
+import { BN } from '@project-serum/anchor'
+import { printBN, printBNtoBN } from '@consts/utils'
+import { DENOMINATOR } from '@invariant-labs/sdk'
+import { PRICE_DECIMAL } from '@consts/static'
+import { ISwap } from '@components/Swap/Swap'
+import { SwapToken } from '@components/Swap/Swap'
+import useStyles from './style'
+
+export const Ido: React.FC<ISwap> = ({ walletStatus, tokens, pools, onSwap, progress }) => {
+  const classes = useStyles()
+  const [swap, setSwap] = React.useState<boolean | null>(null)
+  const [tokenFromIndex, setTokenFromIndex] = React.useState<number | null>(
+    tokens.length ? 0 : null
+  )
+  const [tokenToIndex, setTokenToIndex] = React.useState<number | null>(null)
+  const [amountFrom, setAmountFrom] = React.useState<string>('')
+  const [amountTo, setAmountTo] = React.useState<string>('')
+  const [poolIndex, setPoolIndex] = React.useState<number | null>(null)
+
+  enum feeOption {
+    FEE = 'fee',
+    REVERSED = 'reversed'
+  }
+
+  const calculateSwapOutAmount = (
```

Comment:
why are you using there functions from swap component? also pools have nothing to do with ido component